### ISSUE TITLE
Xournalpp insists on name change to "annotated" for exports

### ISSUE THREAD

**Issue by mjg** [xournalpp/xournalpp#5119]
### Operating System

Linux

### (Linux only) Distribution

Fedora

### (Linux only) Desktop Environment

i3

### (Linux Only) Display Server

X.Org

### Installation Method

git

### Xournal++ Version

1.2.1+dev (00c1af0d)

### libgtk Version

3.24.38

### Bug Description

When exporting to PDF, xournalpp suggests (I would say: changes the name) to `basename_annoted`, where `basename` is the base name of the background PDF, rather than using the base name of the xopp file.

### Expected Behaviour

Use the base name of the xopp file and warn/possible append "annotated" if this would overwrite the background PDF.

### Steps to Reproduce

1. Open PDF
2. Save as xopp
3. Open xopp
4. Export to PDF

### Additional Context

I have not bisected this yet, but it certainly is a change of behaviour. Maybe fallout of the overwrite guards?

xournalpp does not even remember my choice within the same editing session, which is quite a nuisance.

**Comment by bhennion** [comment]
See Preference -> Save/Load where you can customise this.

**Comment by mjg** [comment]
So the main change seems to result from commit 9d6f67d423 which changed the order in which things are tried. In particular, if you have a xopp (named A) with a PDF background (named B), then the template is now filled from B (not A) which is *not* the document name. Was that intentional, or just a side effect?
I would claim that by naming the xopp file, user made an intentional choice for `%{name}` which should not be overriden by whatever is the default.

I have not tracked down since when export names are not remembered - maybe they never were and I didn't notice with the old default.

**Comment by rolandlo** [comment]
I agree that the xopp name (determined by the user) is preferable under usual circumstances. You can have multiple annotations of the same PDF (say multiple student solve the same exercises) and then usually you don't want each export to have the same name.

**Comment by Febbe** [comment]
@bhennion you reviewed that pr which changed this behavior, can you elaborate?

**Comment by bhennion** [comment]
> Was that intentional, or just a side effect?

It was a mistake. Overall, I don't think this wildcards story has been thought through enough. There must be other common use cases that got disturbed.

**Comment by mjg** [comment]
> > Was that intentional, or just a side effect?
> 
> It was a mistake. Overall, I don't think this wildcards story has been thought through enough. There must be other common use cases that got disturbed.

Good, that means we can think about it and change it :)

Right now, if I start xournalpp, save the empty new document, it gets the default name (note/date...), and if I export it to PDF, it gets that name plus "_annotated". I assume we're on the same page that we don't want that. (Note there is no background involved here.)

Same when I open an existing xopp.

If I open a PDF and save the xopp this gets the same base name. If I export that as PDF then we used to have the name conflict with the background PDF, asking to rename the export; this is where the new default is beneficial.

If the new default needs to be configurable we probably need placeholders for xopp basename, main background basename, and a way to tag them with ":overwrite", ":askifexists", ":append" or such to cover different ways of dealing with conflicts.

Related: Is the choice of export name in the dialog remembered (per session)?

**Comment by grubernd** [comment]
Using v1.2.2 on debian.

I found an interesting effect with the mentioned preference setting when trying to use a naming scheme that shows the ancestry of a document. I use that scheme a lot for graphical data where code version control systems are not practical.

Preferences > "Default Pdf export name": `%{name}.xopp`

Expected Behaviour: `name.xopp.pdf`
Current Behaviour: `name.pdf`

I did not open an extra issue because I consider this a sub-variant of "insists on name change". If that is not in line with issue management in the project I apologize beforehand.

**Comment by balkce** [comment]
Just to mention that I'm also interested in fixing this, since I much rather have my exported PDF named the same as the XOPP file. In this manner, I can name my XOPP in whatever manner I want, and the exported PDF file name reflects what I changed in the XOPP file from the original PDF, without modifying the original PDF file name.

In version 1.2.3, the `%{name}` wildcard still refers to the original PDF file name, not the XOPP file name.
